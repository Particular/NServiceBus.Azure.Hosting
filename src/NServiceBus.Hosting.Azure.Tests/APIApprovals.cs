namespace NServiceBus.Hosting.Azure.Tests
{
    using NUnit.Framework;
    using Particular.Approvals;
    using PublicApiGenerator;

    [TestFixture]
    public class APIApprovals
    {
        [Test]
        public void Approve()
        {
            var publicApi = ApiGenerator.GeneratePublicApi(typeof(GenericHost).Assembly);
            Approver.Verify(publicApi);
        }
    }
}