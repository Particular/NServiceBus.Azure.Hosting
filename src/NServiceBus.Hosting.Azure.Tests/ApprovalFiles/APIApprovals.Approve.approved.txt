[assembly: System.Runtime.CompilerServices.InternalsVisibleToAttribute(@"NServiceBus.Hosting.Azure.HostProcess, PublicKey=0024000004800000940000000602000000240000525341310004000001000100dde965e6172e019ac82c2639ffe494dd2e7dd16347c34762a05732b492e110f2e4e2e1b5ef2d85c848ccfb671ee20a47c8d1376276708dc30a90ff1121b647ba3b7259a6bc383b2034938ef0e275b58b920375ac605076178123693c6c4f1331661a62eba28c249386855637780e3ff5f23a6d854700eaa6803ef48907513b92")]
[assembly: System.Runtime.CompilerServices.InternalsVisibleToAttribute(@"NServiceBus.Hosting.Azure.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100dde965e6172e019ac82c2639ffe494dd2e7dd16347c34762a05732b492e110f2e4e2e1b5ef2d85c848ccfb671ee20a47c8d1376276708dc30a90ff1121b647ba3b7259a6bc383b2034938ef0e275b58b920375ac605076178123693c6c4f1331661a62eba28c249386855637780e3ff5f23a6d854700eaa6803ef48907513b92")]
[assembly: System.Runtime.InteropServices.ComVisibleAttribute(false)]
namespace NServiceBus
{
    [System.ObsoleteAttribute("The AsA_Worker role is obsoleted. Manually configure the EndpointConfiguration ob" +
        "ject via IConfigureThisEndpoint.Customize(EndpointConfiguration endpointConfigur" +
        "ation). Will be removed in version 9.0.0.", true)]
    public interface AsA_Worker { }
    public class static ConfigureAzureIntegration
    {
        [System.ObsoleteAttribute("Use code-based configuration instead. The member currently throws a NotImplemente" +
            "dException. Will be removed in version 9.0.0.", true)]
        public static void AzureConfigurationSource(this NServiceBus.EndpointConfiguration config, string configurationPrefix = null) { }
    }
    public class static EndpointConfigurationExtensions
    {
        public static void DefineEndpointName(this NServiceBus.EndpointConfiguration configuration, string endpointName) { }
    }
    [System.ObsoleteAttribute("The azure cloud services host will be deprecated in the next major version. See u" +
        "pgrade guide and documentation for alternatives. Will be treated as an error fro" +
        "m version 9.0.0. Will be removed in version 10.0.0.", false)]
    public class static EndpointStartableAndStoppableExtensions
    {
        public static void RunWhenEndpointStartsAndStops(this NServiceBus.EndpointConfiguration configuration, NServiceBus.IWantToRunWhenEndpointStartsAndStops startableAndStoppable) { }
    }
    public class HostingSettings
    {
        public HostingSettings(string connectionString) { }
        public bool AutoUpdate { get; set; }
        public string Container { get; set; }
        public string LocalResource { get; set; }
        public bool RecycleRoleOnError { get; set; }
        public Microsoft.WindowsAzure.Storage.CloudStorageAccount StorageAccount { get; }
        public int TimeToWaitUntilProcessIsKilled { get; set; }
        public int UpdateInterval { get; set; }
    }
    [System.ObsoleteAttribute("The azure cloud services host will be deprecated in the next major version. See u" +
        "pgrade guide and documentation for alternatives. Will be treated as an error fro" +
        "m version 9.0.0. Will be removed in version 10.0.0.", false)]
    public interface IConfigureThisEndpoint
    {
        void Customize(NServiceBus.EndpointConfiguration configuration);
    }
    [System.ObsoleteAttribute("The azure cloud services host will be deprecated in the next major version. See u" +
        "pgrade guide and documentation for alternatives. Will be treated as an error fro" +
        "m version 9.0.0. Will be removed in version 10.0.0.", false)]
    public interface IConfigureThisHost
    {
        NServiceBus.HostingSettings Configure();
    }
    [System.ObsoleteAttribute("The azure cloud services host will be deprecated in the next major version. See u" +
        "pgrade guide and documentation for alternatives. Will be treated as an error fro" +
        "m version 9.0.0. Will be removed in version 10.0.0.", false)]
    public interface IWantToRunWhenEndpointStartsAndStops
    {
        System.Threading.Tasks.Task Start(NServiceBus.IMessageSession session);
        System.Threading.Tasks.Task Stop(NServiceBus.IMessageSession session);
    }
    [System.ObsoleteAttribute("The azure cloud services host will be deprecated in the next major version. See u" +
        "pgrade guide and documentation for alternatives. Will be treated as an error fro" +
        "m version 9.0.0. Will be removed in version 10.0.0.", false)]
    public class NServiceBusRoleEntrypoint
    {
        public NServiceBusRoleEntrypoint() { }
        public void Start() { }
        public void Stop() { }
    }
}